### DISCUSSION TITLE
Extracting mesh topology from meshing session

### DISCUSSION THREAD

**Discussion by mhoeijm** [ansys/pyfluent#643] [Q&A]
I am working on a workflow where Fluent meshing is used to generate the volume mesh (triangles/tetrahedrons) and I wish to extract the mesh topology from the mesher for later use in LS-DYNA.  Currently I write the mesh file to disk and read in the .msh.h5 file using `h5py` and re-construct the mesh myself using `NumPy`.

Information I would like to extract from a PyFluent session include: 
- node coordinates
- element/cell definitions (e.g. for tetrahedrons. n1, n2, n3, n4)
- face and face-zone definitions and names

Are there any plans on exposing methods to get mesh topology from a PyFluent session? Or is this out of scope for PyFluent?

**Comment by dnwillia-work** [comment]
@ansmhoeij we don't have plans in the works for this currently though you can extract the surface meshes once you have switched to solve mode using the field data module.

If I've understood correctly though what you are trying to do is use Fluent meshing to generate a mesh that is later consumed into LS-Dyna, is that right?

**Reply by mhoeijm** [reply]
@dnwillia-work Yes that is correct. I am interested in just using Fluent meshing to create the volume mesh. I noticed that in solve mode you can extract the surface. However, I would need to have the interior cells/elements as well.

> you can extract the surface meshes once you have switched to solve mode using the field data module.

I tried this but unfortunately I am getting a gRPC-related error:

```Python
    session = pyfluent.launch_fluent(start_instance=True, show_gui=False, meshing_mode=True)

    session.meshing.tui.file.read_mesh(
        "volume_mesh.msh.h5"
    )
    session.meshing.tui.switch_to_solution_mode("yes")
    field_data = session.field_data
    field_data.add_get_surfaces_request(
        surface_ids=[10296], provide_vertices=True, provide_faces=True
    )
    payload_data = field_data.get_fields()
```

![image](https://user-images.githubusercontent.com/102799582/179770041-3e7ad660-e865-4c0d-a852-4e72ef7e093b.png)

**Reply by dnwillia-work** [reply]
Yeah, ouch.  Please file an issue for that one.  I'm not certain why it's not working.

Regarding the mesh files I would think for now you can export the mesh from Fluent into something LS-Dyna can read?  I'd have to look at the documentation to figure it out.
